src/main/java/fr/insalyon/citi/golo/compiler/JavaBytecodeEnumGenerator.java

```diff
@@ -0,0 +1,267 @@
+    }
+    visitor.visitInsn(ARETURN);
+    visitor.visitMaxs(0, 0);
+    visitor.visitEnd();
+  }
+
+  private CodeGenerationResult makeEnumValue(ClassWriter parentClassWriter, String sourceFilename, GoloEnum.Value value) {
+    String enumType = value.getEnum().getPackageAndClass().toJVMType();
+    String valueType = value.getPackageAndClass().toJVMType();
+    ClassWriter classWriter = new ClassWriter(COMPUTE_FRAMES | COMPUTE_MAXS);
+    classWriter.visitSource(sourceFilename, null);
+    classWriter.visit(V1_8, ACC_PUBLIC | ACC_SUPER | ACC_FINAL, valueType, null, enumType, null);
+    classWriter.visitInnerClass(valueType, enumType, value.getName(), ACC_PUBLIC | ACC_FINAL | ACC_STATIC);
+    parentClassWriter.visitInnerClass(valueType, enumType, value.getName(), ACC_PUBLIC | ACC_FINAL | ACC_STATIC);
+    for (String member : value.getMembers()) {
+      classWriter.visitField(ACC_PUBLIC | ACC_FINAL, member, "Ljava/lang/Object;", null, null).visitEnd();
+    }
+    if (value.hasMembers()) {
+      makeValuedConstructor(classWriter, value);
+      makeHashCode(classWriter, value);
+      makeEquals(classWriter, value);
+    } else {
+      makeDefaultConstructor(classWriter, enumType);
+      parentClassWriter.visitField(ACC_PUBLIC | ACC_FINAL | ACC_STATIC, value.getName(),
+            value.getEnum().getPackageAndClass().toJVMRef(), null, null).visitEnd();
+    }
+    makeToString(classWriter, value);
+    classWriter.visitEnd();
+    return new CodeGenerationResult(classWriter.toByteArray(), value.getPackageAndClass());
+  }
+
+  private void makeEquals(ClassWriter cw, GoloEnum.Value value) {
+    String target = value.getPackageAndClass().toJVMType();
+    MethodVisitor mv = cw.visitMethod(ACC_PUBLIC, "equals", "(Ljava/lang/Object;)Z", null, null);
+    Label notSameInstance = new Label();
+    Label notNull = new Label();
+    Label sameType = new Label();
+    Label allAttrsEquals = new Label();
+    Label attrNotEqual = new Label();
+    mv.visitCode();
+
+    // if (other == this) { return true; }
+    mv.visitVarInsn(ALOAD, 1);
+    mv.visitVarInsn(ALOAD, 0);
+    mv.visitJumpInsn(IF_ACMPNE, notSameInstance);
+    mv.visitInsn(ICONST_1);
+    mv.visitInsn(IRETURN);
+    mv.visitLabel(notSameInstance);
+
+    // if (other == null) { return false; }
+    mv.visitVarInsn(ALOAD, 1);
+    mv.visitJumpInsn(IFNONNULL, notNull);
+    mv.visitInsn(ICONST_0);
+    mv.visitInsn(IRETURN);
+    mv.visitLabel(notNull);
+
+    // if (!(other instanceof <value>)) { return false; }
+    mv.visitVarInsn(ALOAD, 1);
+    mv.visitTypeInsn(INSTANCEOF, target);
+    mv.visitJumpInsn(IFNE, sameType);
+    mv.visitInsn(ICONST_0);
+    mv.visitInsn(IRETURN);
+    mv.visitLabel(sameType);
+
+    // cast other to <value>
+    mv.visitVarInsn(ALOAD, 1);
+    mv.visitTypeInsn(CHECKCAST, target);
+    mv.visitVarInsn(ASTORE, 2);
+
+    // java.util.Objects.equals(<member>, other.<member>)
+    for (String member : value.getMembers()) {
```

Comment:
`Objects.equals` is used here to compare the members, while the checks above tests for the instance itself